pallene/translator.lua

```diff
@@ -46,65 +48,105 @@ function Translator:add_whitespace(input, start_index, stop_index)
-                end
-            end
-        elseif node._tag == "ast.Toplevel.Func" then
-            -- Remove type annotations from function parameters.
-            for _, arg_decl in ipairs(node.value.arg_decls) do
-                -- Type annotations are mandatory for function parameters.
-                -- Remove the colon but retain any adjacent comment to the right.
-                instance:add_whitespace(input, arg_decl.col_loc.pos, arg_decl.col_loc.pos)
-                -- Remove the type annotation but exclude the next token.
-                instance:add_whitespace(input, arg_decl.type.loc.pos, arg_decl.end_loc.pos - 1)
+function Translator:add_exports()
+    if #self.exports > 0 then
+        table.insert(self.partials, "\nreturn {\n")
+        for _, export in ipairs(self.exports) do
+            local pair = string.format("    %s = %s,\n", export, export)
+            table.insert(self.partials, pair)
+        end
+        table.insert(self.partials, "}\n")
+    end
+end
+
+function Translator:translate_decl(decl)
+    if decl.type then
+        -- Remove the colon but retain any adjacent comment to the right.
+        self:add_whitespace(decl.col_loc.pos, decl.col_loc.pos)
+        -- Remove the type annotation but exclude the next token.
+        self:add_whitespace(decl.type.loc.pos, decl.end_loc.pos - 1)
+    end
+end
+
+function Translator:translate_stat(block)
+    for _, statement in ipairs(block.stats) do
+        if statement._tag == "ast.Stat.Decl" then
+            for _, decl in ipairs(statement.decls) do
+                self:translate_decl(decl)
             end
+        elseif statement._tag == "ast.Stat.For" then
+            self:translate_decl(statement.decl)
+        elseif statement._tag == "ast.Stat.Block" then
+            self:translate_stat(statement)
+        end
+    end
+end
 
-            -- Remove type annotations from the return type, which is optional. However, `rt_col_loc`
-            -- and `rt_end_loc` are always set. Therefore, it is safe to replace without any checks.
-            instance:add_whitespace(input, node.rt_col_loc.pos, node.rt_end_loc.pos - 1)
-
-            -- Remove type annotations from local declarations.
-            for _, statement in ipairs(node.value.body.stats) do
-                if statement._tag == "ast.Stat.Decl" then
-                    for _, decl in ipairs(statement.decls) do
-                        if decl.type then
-                            -- Remove the colon but retain any adjacent comment to the right.
-                            instance:add_whitespace(input, decl.col_loc.pos, decl.col_loc.pos)
-                            -- Remove the type annotation but exclude the next token.
-                            instance:add_whitespace(input, decl.type.loc.pos, decl.end_loc.pos - 1)
-                        end
-                    end
-                end
+function Translator:translate_tl(node)
```

Comment:
To be consistent with the other files, please call this function `translate_toplevel`.